src/main/java/dev/lu15/voicechat/network/voice/VoiceServer.java

```diff
@@ -232,25 +230,81 @@ private void handle(@NotNull Player player, @NotNull YouHereBroPacket ignored) {
+            });
+        } else {
+            Group group = groupManager.getGroup(player);
+            switch (group.type()) {
+                case NORMAL, ISOLATED -> {
+                    EventDispatcher.callCancellable(event, () -> {
+                        GroupSoundPacket soundPacket = new GroupSoundPacket(
+                                player.getUuid(), // the channel is the sender's UUID
+                                player.getUuid(),
+                                event.getAudio(),
+                                packet.sequenceNumber(),
+                                SoundSources.GROUP
+                        );
+                        groupManager.getPlayers(group).stream().filter(p -> {
+                            if (p.equals(player)) return false;
+                            return !p.hasTag(Tags.PLAYER_STATE) || !p.getTag(Tags.PLAYER_STATE).disabled();
+                        }).forEach(p -> this.write(p, soundPacket));
+                    });
+                }
+                case OPEN -> {
+                    EventDispatcher.callCancellable(event, () -> {
+                        PlayerSoundPacket soundPacketPlayer = new PlayerSoundPacket(
+                                player.getUuid(), // the channel is the sender's UUID
+                                player.getUuid(),
+                                event.getAudio(),
+                                packet.sequenceNumber(),
+                                event.getSoundSelector().distance(),
+                                packet.whispering(),
+                                SoundSources.PROXIMITY
+                        );
+                        GroupSoundPacket soundPacketGroup = new GroupSoundPacket(
+                                player.getUuid(), // the channel is the sender's UUID
+                                player.getUuid(),
+                                event.getAudio(),
+                                packet.sequenceNumber(),
+                                SoundSources.GROUP
+                        );
+
+                        List<Player> groupPlayers = groupManager.getPlayers(group).stream().filter(p -> {
+                            if (p.equals(player)) return false;
+                            return !p.hasTag(Tags.PLAYER_STATE) || !p.getTag(Tags.PLAYER_STATE).disabled();
+                        }).toList();
+                        List<Player> players = event.getSoundSelector().canHear(player).stream().filter(p -> {
+                            if (p.equals(player)) return false;
+                            if(groupPlayers.contains(p)) return false;
+                            VoiceState vs = p.getTag(Tags.PLAYER_STATE);
+                            if(vs!=null&&vs.group()!=null&&groupManager.getGroup(vs.group()).type()==Group.Type.ISOLATED) return false;
+                            return !p.hasTag(Tags.PLAYER_STATE) || !p.getTag(Tags.PLAYER_STATE).disabled();
+                        }).toList();
```

Comment:
Is this the same code as above? This could probably be turned into another method.